### PR TITLE
Enable the HWIntrinsic extension methods and remove the instance implementations

### PR DESCRIPTION
**Pull request by tannergooding**
The resolves https://github.com/dotnet/corefx/issues/34902 by finishing out the transition of the various HWIntrinsic helper methods to be extension methods, rather than instance ones.

CC. @fiigii, @CarolEidt

### CODE REVIEW

**Comment by fiigii** [comment]
Oh, there are other assert failures in CI tests.

**Comment by tannergooding** [comment]
Had to do a bit of additional cleanup, part of which involved resolving https://github.com/dotnet/coreclr/issues/22537.

**Comment by tannergooding** [comment]
There is also something broken about calling the various `As` methods via reflection when `TFrom` and `TTo` don't match up. They end up failing with:
```
Assert failure(PID 24828 [0x000060fc], Thread: 7300 [0x1c84]): Assertion failed 'src->gtObj.gtClass == structHnd' in 'System.Runtime.Intrinsics.Vector128:AsDouble(struct):struct' (IL size 12)

    File: E:\Users\tagoo\Repos\coreclr\src\jit\importer.cpp Line: 1314
    Image: E:\Users\tagoo\Repos\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root\CoreRun.exe
```

I'm still digging in, but perhaps @CarolEidt has some insight here?

File: src/jit/importer.cpp [hunk 1]
```diff
@@ -5188,25 +4485,32 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                                 result = NI_Base_Vector256_AsUInt64;
                             }
                         }
-                        else if (strcmp(methodName, "get_Zero") == 0)
+                        else if (strcmp(methodName, "CreateScalarUnsafe") == 0)
```
Comment (fiigii) [inline src/jit/importer.cpp]: I think we should unify the Vector intrinsic name lookup code with other hardware intrinsics that using the table-driven framework. The current big if-sequence seems not worthy for perf. And more important, we have discussed several times that we want to improve the intrinsic recognition perf, so a unified name lookup is desired.
But that could be a follow-up PR.

File: src/jit/hwintrinsicxarch.cpp [hunk 2]
```diff
@@ -952,6 +955,641 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
...
+        case NI_Base_Vector128_AsUInt32:
+        case NI_Base_Vector128_AsUInt64:
+        {
+            // We fold away the cast here, as it only exists to satisfy
+            // the type system. It is safe to do this here since the retNode type
+            // and the signature return type are both the same TYP_SIMD.
+
+            assert(sig->numArgs == 1);
+
+            retNode = impSIMDPopStack(retType, /* expectAddr: */ false, sig->retTypeClass);
```
Comment (tannergooding) [inline src/jit/hwintrinsicxarch.cpp]: I had dropped the `structType: sign->retTypeClass` when moving the code over, pretty tsure this was the issue with https://github.com/dotnet/coreclr/pull/22705#issuecomment-465717622

Comment (tannergooding) [reply src/jit/hwintrinsicxarch.cpp]: Confirmed locally that this fixed it.

File: src/jit/importer.cpp (hunk 1 above)
Comment (tannergooding) [reply src/jit/importer.cpp]: Agreed. I'll submit a follow up PR once this one is merged.

**Comment by tannergooding** [comment]
Had to push a new commit as I missed porting the `Base_Vector128_As*` intrinsics into the arm64 specific code.

File: src/jit/hwintrinsicxarch.cpp [hunk 3]
```diff
@@ -952,6 +955,641 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
...
+        return nullptr;
+    }
+
+    unsigned  simdSize = 0;
+    var_types baseType = TYP_UNKNOWN;
+    var_types retType  = JITtype2varType(sig->retType);
+
+    assert(!sig->hasThis());
+
+    if (HWIntrinsicInfo::BaseTypeFromFirstArg(intrinsic))
```
Comment (fiigii) [inline src/jit/hwintrinsicxarch.cpp]: We can let `impIsTableDrivenHWIntrinsic` return `true` for `Base` to eliminate this code block.

**Pull request merged**